care/facility/api/viewsets/patient.py

```diff
@@ -279,6 +283,52 @@ def filter_by_diagnoses(self, queryset, name, value):
             )
         return queryset.filter(filter_q)
 
+    last_consultation__consent_types = MultiSelectFilter(
+        method="filter_by_has_consents"
+    )
+
+    def filter_by_has_consents(self, queryset, name, value: str):
+
+        if not value:
+            return queryset
+
+        values = value.split(",")
+
+        filter_q = Q()
+
+        consultation_consent_files = (
+            FileUpload.objects.filter(
+                file_type=FileUpload.FileType.CONSENT_RECORD,
+                is_archived=False,
+            )
+            .order_by("associating_id")
+            .values_list("associating_id", flat=True)
+        )
+
+        consultations = PatientConsent.objects.filter(
+            external_id__in=[x for x in consultation_consent_files],
```

Comment:
This is way too expensive, it loads up all consent records all the time, its easier to set a flag for a patient/consultation and then use that.

Reply:
Can we merge this as it is P1 and I create a follow up PR with the flag?

Reply:
atleast move it as a  subquery to optimise it a bit, the current version is extremely unoptimised.

Reply:
I have gone ahead and replaced the `has_consents` boolean field to an array field, that way we don't have to recalculate on the filter. Need your review